File: EaterSpring/src/main/kotlin/xyz/purema/eater/models/OrderItem.kt
package xyz.purema.eater.models

import java.util.*
import javax.persistence.Entity
import javax.persistence.Id
import javax.persistence.ManyToOne

@Entity
data class OrderItem(
    @ManyToOne
    val productOrder: ProductOrder,
    val productId: String,
    val name: String,
    val price: Long,
    val qty: Int
) {
    @Id
    val id: String = UUID.randomUUID().toString()
}

File: EaterNet/Eater/Models/OrderItem.cs
namespace Eater.Models
{
    public class OrderItem
    {
        public string Id { get; set; } = Guid.NewGuid().ToString();
        //public ProductOrder? ProductOrder { get; set; }
        public string ProductId { get; set; }
        public string Name { get; set; }
        public long Price { get; set; }
        public int Qty { get; set; }

        public OrderItem(string productId, string name, long price, int qty)
        {
            ProductId = productId;
            Name = name;
            Price = price;
            Qty = qty;
        }
    }
}
